SQLFluff fails when comparing two identical fixes

I'm guessing this is due to @jpy-git 's recent linting improvements.  This is coming up in the checks of my most recent PR, #2130.

## Expected Behaviour

PR checks do not fail with a LintFix object attribute error.

## Observed Behaviour

https://github.com/sqlfluff/sqlfluff/runs/4571921276?check_suite_focus=true

```
self = <LintFix: delete @[L:  3, P: 16] delete:'    '>
other = <LintFix: delete @[L:  3, P: 16] delete:'    '>

    def __eq__(self, other):
        """Compare equality with another fix.
    
        A fix is equal to another if is in the same place (position), with the
        same type and (if appropriate) the same edit values.
    
        """
        if not self.edit_type == other.edit_type:
            return False
        if not self.anchor == other.anchor:
            return False
>       if not self.edit == other.edit:
E       AttributeError: 'LintFix' object has no attribute 'edit'
```

--- a/src/sqlfluff/core/rules/base.py
+++ b/src/sqlfluff/core/rules/base.py
@@ -130,6 +130,7 @@ class LintFix:
         if not anchor:  # pragma: no cover
             raise ValueError("Fixes must provide an anchor.")
         self.anchor = anchor
+        self.edit: Optional[List[BaseSegment]] = None
         if edit is not None:
             # Coerce edit iterable to list
             edit = list(edit)
@@ -153,8 +154,6 @@ class LintFix:
             # Once stripped, we shouldn't replace any markers because
             # later code may rely on them being accurate, which we
             # can't guarantee with edits.
-        else:
-            self.edit = None
 
     def is_trivial(self):
         """Return true if the fix is trivial.

--- a/src/sqlfluff/core/rules/base.py
+++ b/src/sqlfluff/core/rules/base.py
@@ -153,6 +153,8 @@ class LintFix:
             # Once stripped, we shouldn't replace any markers because
             # later code may rely on them being accurate, which we
             # can't guarantee with edits.
+        else:
+            self.edit = None
 
     def is_trivial(self):
         """Return true if the fix is trivial.

